TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ViperArrrrrm.java

```diff
@@ -30,35 +26,33 @@ void home(){
     }
     void highBucket(){
         viperArm.setPower(1);
-        viperArm.setTargetPosition(11300);
+        viperArm.setTargetPosition(3000);
     }
     void lowBucket(){
         viperArm.setPower(1);
-        viperArm.setTargetPosition(7000);
+        viperArm.setTargetPosition(1700);
     }
 
     void operateArm(Telemetry telemetry, double armPower){
-//        double max = 1;
-//        max = Math.max(max, Math.abs(armPower));
-//        if (armPower > max) {
-//            armPower = max;
-//        }
-//
-//        if (touchSensor.isPressed() && armPower < 0) {
-//            armPower = 0;
-//            viperArm.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
-//            telemetry.addData("Deb", "Pressed");
-//        } else {
-//            telemetry.addData("Deb", "Not Pressed");
-//        }
-//        //upper limit
-//        if (viperArm.getCurrentPosition() >= 11300 && armPower > 0){ //this is viper max height
-//            armPower = 0;
-//            telemetry.addData("Deb", "Upper Limit");
-//        }
-//
-//        viperArm.setPower(armPower);
-//        telemetry.addData("arm power", "%4.2f", armPower);
+
+        if ((viperArm.getTargetPosition() <= 0) && (viperArm.getCurrentPosition() <= 0)) {
+            if (touchSensor.getValue() == 1) {
+                viperArm.setTargetPosition(0);
+                viperArm.setPower(0);
+                viperArm.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+                viperArm.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+                viperArm.setPower(1);
+            } else {
+                viperArm.setTargetPosition(viperArm.getCurrentPosition() - 100);
+                viperArm.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+                viperArm.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+                viperArm.setPower(1);
+            }
+        }
```

Comment:
Looking at this, I'm noticing that we'll never stop 'early' - that is, if the motor slips so that the bucket is _lower_ than we think, we'll keep driving the motor until the encoder gets to '0', and only then test to see if the button is being pressed. I think it might be better to move the check for `viperArm.getCurrentPosition() <= 0`, like this:

```suggestion
        if (viperArm.getTargetPosition() <= 0) {
            if (touchSensor.getValue() == 1) {
                viperArm.setTargetPosition(0);
                viperArm.setPower(0);
                viperArm.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
                viperArm.setMode(DcMotor.RunMode.RUN_TO_POSITION);
                viperArm.setPower(1);
            } else if (touchSensor.getValue == 0 && viperArm.getCurrentPosition() <= 0) {
                viperArm.setTargetPosition(viperArm.getCurrentPosition() - 100);
                viperArm.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
                viperArm.setMode(DcMotor.RunMode.RUN_TO_POSITION);
                viperArm.setPower(1);
            }
        }
```
Does that make sense to you?